falcon/asgi/app.py

```diff
@@ -1025,29 +1026,41 @@ def _prepare_middleware(self, middleware=None, independent_middleware=False):
             asgi=True,
         )
 
-    async def _http_status_handler(self, req, resp, status, params):
-        self._compose_status_response(req, resp, status)
+    async def _http_status_handler(self, req, resp, status, params, ws=None):
+        if resp:
+            self._compose_status_response(req, resp, status)
+        else:
+            assert ws
```

Comment:
One we are done debugging, IMHO we ought to remove these `assert`s. It is possible to run the interpreter with ignoring `assert` statements anyway

Reply:
I kinda like there as a way of documenting the use of the function.

I can use an `elif we:` and an actual raise there.

Reply:
Yeah probably `elif ws:` is better I think

Reply:
ok before it was `if ws:`, but I think a proper error should be raised, to indicate that it's called with either a resp or a ws

Reply:
Yeah I think you current approach looks really nice, it is zero-cost if not reached, but still providing meaningful info for debugging edge cases. :+1: